File: src/models/Order.java
package models;

import java.util.Map;

public class Order {
    private int orderID;
    private User user;
    private Map<Product, Integer> items;
    private String status;

    public Order(int orderID, User user, Map<Product, Integer> items) {
        this.orderID = orderID;
        this.user = user;
        this.items = items;
        this.status = "Processing";
    }

    public void placeOrder() {
        for (Map.Entry<Product, Integer> entry : items.entrySet()) {
            entry.getKey().updateStock(entry.getValue());
        }
        user.addOrder(this);
    }

    public void updateStatus(String newStatus) {
        this.status = newStatus;
    }

    public int getOrderID() { return orderID; }
    public String getStatus() { return status; }
}

File: backend/order.py
class Order:
    def __init__(self, order_id, user, items):
        self._order_id = order_id
        self._user = user
        self._items = items
        self._status = "en attente"

    def place_order(self):
        for product, qty in self._items:
            product.update_stock(qty)
        self._user._order_history.append(self)

    def update_status(self, new_status):
        self._status = new_status
